src/main/java/net/replaceitem/integratedcircuit/network/ServerPacketHandler.java

```diff
@@ -29,8 +32,36 @@ public static void receiveFinishEditingPacket(FinishEditingC2SPacket packet, Ser
         player.updateLastActionTime();
         ServerWorld serverWorld = player.getServerWorld();
         BlockEntity blockEntity = serverWorld.getBlockEntity(packet.pos());
-        if (!(blockEntity instanceof IntegratedCircuitBlockEntity integratedCircuitBlockEntity)) return;
-        integratedCircuitBlockEntity.removeEditor(player);
+
+        if (blockEntity instanceof IntegratedCircuitBlockEntity integratedCircuitBlockEntity) {
+            integratedCircuitBlockEntity.removeEditor(player);
+        }
+    }
+
+    public static void receiveRenameCircuitPacket(RenameCircuitC2SPacket packet, ServerPlayNetworking.Context context) {
+        BlockPos pos = packet.pos();
+        ServerPlayerEntity renamingPlayer = context.player();
+        ServerWorld serverWorld = renamingPlayer.getServerWorld();
+        BlockEntity blockEntity = serverWorld.getBlockEntity(pos);
+
+        renamingPlayer.updateLastActionTime();
+
+        if (blockEntity instanceof IntegratedCircuitBlockEntity integratedCircuitBlockEntity) {
+
+            Text currentName = integratedCircuitBlockEntity.getCustomName();
+            Text newName = packet.newName();
+
+            if (!Objects.equals(currentName, newName)) {
+                integratedCircuitBlockEntity.setCustomName(newName);
+
+                for (ServerPlayerEntity player : serverWorld.getPlayers()) {
```

Comment:
I think this does not need to be sent to all other players. Doing `integratedCircuitBlockEntity.setCustomName(newName)` should make minecraft send the block entity update itself (due to markDirty). Only the players who currently have the circuit screen open might need an explicit update (?). For that you can use `IntegratedCircuitBlockEntity.editors`